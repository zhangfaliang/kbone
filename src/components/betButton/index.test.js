import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import BetButton from './index';

describe('BetButton', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<BetButton />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('prefixCls null', () => {
    expect.assertions(1);
    const wrapper = render(<BetButton prefixCls={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('handleClick', () => {
    expect.assertions(1);
    const button = mount(<BetButton />);
    const clickCheckBtn = button.prop('clickBetBtn');
    const click = jest.fn(clickCheckBtn);
    button.simulate('click');
    expect(click()).toBe('clickBetBtn');
  });

  it('changeSp', () => {
    expect.assertions(2);
    const button = mount(<BetButton />);
    button.setProps({
      sp: 1.5,
    });
    const spUpFlag = button
      .children('div')
      .at(0)
      .children('span')
      .at(0)
      .hasClass('flickerSp');
    expect(1.5).toBe(button.prop('sp'));
    button.setProps({
      sp: 1.1,
    });
    const downFlag = button
      .children('div')
      .at(0)
      .children('span')
      .at(0)
      .hasClass('up');
    expect(1.1).toBe(button.prop('sp'));
  });

  it('changeHandicap', () => {
    expect.assertions(1);
    const button = mount(<BetButton />);
    button.setProps({
      handicap: 5.9,
    });
    expect(5.9).toBe(button.prop('handicap'));
  });

  it('disabledBtn', () => {
    expect.assertions(1);
    const button = render(<BetButton disabled={true} />);
    expect(toJson(button)).toMatchSnapshot();
  });

  it('hanclick-disabledBtn', () => {
    expect.assertions(2);
    const clickCheckBtn = () => 'clickBetBtn';
    const click = jest.fn(clickCheckBtn);
    const button = mount(<BetButton clickCheckBtn={click} disabled={!!1} />);
    button.simulate('click');
    expect(click).not.toHaveReturned();
    expect(click).not.toHaveReturnedWith('clickBetBtn');
  });
});
