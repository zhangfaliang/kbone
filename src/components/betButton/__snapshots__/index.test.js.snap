// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetButton base 1`] = `
<div
  class="default-bet-button"
>
  <span
    class="up"
  >
    <span
      class="iconfont icon-sheng"
    />
  </span>
  <span
    class="down"
  >
    <span
      class="iconfont icon-jiang"
    />
  </span>
  <div
    class="wrapSp"
  >
    <span
      class="sp"
    />
  </div>
</div>
`;

exports[`BetButton disabledBtn 1`] = `
<div
  class="default-bet-button disabled"
>
  <span
    class="up"
  >
    <span
      class="iconfont icon-sheng"
    />
  </span>
  <span
    class="down"
  >
    <span
      class="iconfont icon-jiang"
    />
  </span>
  <div
    class="wrapSp"
  >
    <span
      class="lock"
    >
       
      <span
        class="iconfont icon-lock"
      />
    </span>
  </div>
</div>
`;

exports[`BetButton prefixCls null 1`] = `
<div
  class="default-bet-button"
>
  <span
    class="up"
  >
    <span
      class="iconfont icon-sheng"
    />
  </span>
  <span
    class="down"
  >
    <span
      class="iconfont icon-jiang"
    />
  </span>
  <div
    class="wrapSp"
  >
    <span
      class="sp"
    />
  </div>
</div>
`;
